### PR TITLE
Modal refactor, please review and merge

### PR DESCRIPTION
**Pull request by Samcraig77**

### CODE REVIEW

File: index.css [hunk 1]
```diff
@@ -206,23 +206,33 @@ hr {
...
+    position: fixed;
+    top: 0;
+    bottom:0;
+    left: 0;
+    right: 0;
+    width: 100vw;
+    height: 100vh;
+    background-color: transparent;
+    z-index: 1;
+
```
Comment (Raehale) [inline index.css:220]: Could you remove this extra space

File: index.css [hunk 2]
```diff
@@ -206,23 +206,33 @@ hr {
 
 /***MODAL***/
 /**SIGN UP**/
+
+.modal-wrapper{
```
Comment (Raehale) [inline index.css:210]: Could you add a space after the name and before the {

File: index.css [hunk 3]
```diff
@@ -206,23 +206,33 @@ hr {
 
 /***MODAL***/
 /**SIGN UP**/
+
```
Comment (Raehale) [inline index.css:209]: Could you remove this space

File: index.css [hunk 4]
```diff
@@ -206,23 +206,33 @@ hr {
...
+.modal-wrapper{
+    position: fixed;
+    top: 0;
+    bottom:0;
+    left: 0;
+    right: 0;
+    width: 100vw;
+    height: 100vh;
```
Comment (Raehale) [inline index.css:217]: typically the height should be set to a min-height so if you have extra content it doesn't cut off

File: index.css [hunk 5]
```diff
@@ -206,23 +206,33 @@ hr {
...
     position: fixed;
     top: 50%;
     left: 50%;
     transform: translate(-50%, -50%);
-
-    display: none; /*grid when displayed*/
-
+    display: grid;
     color: var(--text);
-
```
Comment (Raehale) [inline index.css:218]: I add these spaces for ease of reading, lets have a design talk about this

File: index.css [hunk 6]
```diff
@@ -453,7 +468,3 @@ select.modal-question {
     }
 
 }
-
-.hidden {
-    display: none;
-}
```
Comment (Raehale) [inline index.css:459]: github prefers when there's an empty line at the end of each file (I think its bc of c++) but could you add that?

File: index.js [hunk 7]
```diff
@@ -48,55 +48,37 @@ const loginBtnsEl = document.getElementById("login-btns")
...
-        const currentModal = event.target.parentElement
-
-        hideModal(currentModal)
-
-        document.querySelectorAll(".login-btn").forEach(function(loginBtn) {
-            enableBtn(loginBtn)
-        })
-    })
+document.addEventListener("click", event => {
+    [...event.target.classList].includes('x-out') ? event.target.parentElement.parentElement.classList.toggle('hidden') :
```
Comment (Raehale) [inline index.js:62]: love this! Didn't know you could spread it out like that, or rather never considered.

File: index.js [hunk 8]
```diff
@@ -48,55 +48,37 @@ const loginBtnsEl = document.getElementById("login-btns")
...
     toggleLogInOutBtns()
 
     loggedIn = true
 })
 
-loginBtn.addEventListener("click", function(event) {
+loginBtn.addEventListener("click", function() {
     enableBtns()
-    hideModal(event.target.parentElement.parentElement)
+    displayGenericModal('sign-in')
```
Comment (Raehale) [inline index.js:81]: This is an issue that I think I introduced but we should make sign-in and log-in one or the other

**Review by Raehale — CHANGES_REQUESTED** [review]
Amazing job! You right code super cleanly and have unique approaches I always love seeing.

**Comment by Samcraig77** [comment]
Rae did you mean to merge this? I'm on my phone and can't make the changes right now but I will look at them tomorrow :)

**Comment by Raehale** [comment]
I did. The changes were super tiny, cosmetic. It'd be cool if you could make them but I wanted to work off you and Vicentes work. Thanks so much! And don't worry about the timeline I know weekends are busy.

**Pull request merged**